### PR TITLE
Implement ExecSync api

### PR DESCRIPTION
**Pull request by Crazykev**
1. move streaming related api (#41) to a separate file
2. vendor `kubelet/util/ioutils` and `docker/pkg/stdcopy` to implement io with hyper container.
3. implement(actually copy from hyperhq/hyper) stdout extractor for hyper
4. implement ExecSync api releated code.
5. add test case test for exec success and failure.

PTAL @feiskyer

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/frakti/47)

### CODE REVIEW

**Comment by Crazykev** [comment]
@feiskyer Conflict resolved.

File: pkg/hyper/client.go [hunk 1]
```diff
@@ -364,3 +361,164 @@ func (c *Client) CreateContainer(podID string, spec *types.UserContainer) (strin
...
+
+	resp, err := c.client.Wait(ctx, &req)
+	if err != nil {
+		return -1, err
+	}
+
+	return resp.ExitCode, nil
+}
+
+func (c *Client) CheckContainerRunningStatus(containerID string) error {
```
Comment (feiskyer) [inline pkg/hyper/client.go]: rename it to `IsContainerRunning` for clear.

File: pkg/hyper/extractor.go [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+package hyper
+
+import (
+	"encoding/binary"
+	"fmt"
+
+	"github.com/docker/docker/pkg/stdcopy"
+)
+
+type StreamExtractor interface {
```
Comment (feiskyer) [inline pkg/hyper/extractor.go]: Please un-export those methods and types, also add docs for methods and type definitions in this file.

File: pkg/hyper/client.go (hunk 1 above)
Comment (Crazykev) [inline pkg/hyper/client.go]: @feiskyer `IsContainerRunning` sounds like should return a bool value, this method is mainly for check if container is running, if not or get status failed, return error. So does `CheckIfContainerRunning` sound better to you?

File: pkg/hyper/extractor.go (hunk 2 above)
Comment (Crazykev) [inline pkg/hyper/extractor.go]: Indeed should have one, later we can add that docs to hyper, too.

File: pkg/hyper/client.go (hunk 1 above)
Comment (feiskyer) [inline pkg/hyper/client.go]: LGTM

**Comment by Crazykev** [comment]
@feiskyer Updated.

**Comment by feiskyer** [comment]
LGTM. Thanks.

**Pull request merged**